### PR TITLE
Prevent PSBT::extract_tx from panicing

### PR DESCRIPTION
**Pull request by stevenroose**

### CODE REVIEW

**Comment by stevenroose** [comment]
I think this is better than an extra Error variant because it's named `extract_tx`, which doesn't make claims on the finality of the tx.

File: src/util/psbt/mod.rs [hunk 1]
```diff
@@ -58,13 +59,14 @@ impl PartiallySignedTransaction {
...
-    /// Extract a Transaction from a finalized PartiallySignedTransaction
+    /// Extract the Transaction from a PartiallySignedTransaction by filling 
+    /// in the available signature information in place.
     pub fn extract_tx(self) -> Transaction {
         let mut tx: Transaction = self.global.unsigned_tx;
 
         for (vin, psbtin) in tx.input.iter_mut().zip(self.inputs.into_iter()) {
-            vin.script_sig = psbtin.final_script_sig.unwrap();
-            vin.witness = psbtin.final_script_witness.unwrap();
+            vin.script_sig = psbtin.final_script_sig.or_else(|| Some(Script::new())).unwrap();
```
Comment (dongcarl) [inline src/util/psbt/mod.rs]:
```suggestion
            vin.script_sig = psbtin.final_script_sig.unwrap_or_else(|| Script::new());
```

File: src/util/psbt/mod.rs [hunk 2]
```diff
@@ -58,13 +59,14 @@ impl PartiallySignedTransaction {
...
+            vin.witness = psbtin.final_script_witness.or_else(|| Some(Vec::new())).unwrap();
```
Comment (dongcarl) [inline src/util/psbt/mod.rs]:
```suggestion
            vin.witness = psbtin.final_script_witness.unwrap_or_else(|| Vec::new());
```

**Review by dongcarl — CHANGES_REQUESTED** [review]

**Comment by stevenroose** [comment]
Hehe, fine. Didn't know about those methods.

**Review by dongcarl — APPROVED** [review]

**Pull request merged**